### PR TITLE
PRL-6620: Fix submitting refuge without refuge document

### PR DESCRIPTION
**Pull request by hcarslaw**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-6620

### Change description ###
Fixes ability to submit refuge without a refuge document by using the back button.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts [hunk 1]
```diff
@@ -545,7 +553,11 @@ export const ApplicantDetails = (
     if (sessionApplicantData[applicant]['liveInRefuge'] === YesOrNo.YES) {
       newApplicantData.push({
         key: keys['c8RefugeDocument'],
-        value: sessionApplicantData[applicant]['refugeConfidentialityC8Form']?.['document_filename'],
+        keyName: `${sessionApplicantData[applicant]['applicantFirstName']}${sessionApplicantData[applicant]['applicantLastName']}c8RefugeDocument`,
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts:556]: Can have the static names

File: src/main/steps/c100-rebuild/check-your-answers/content.ts [hunk 2]
```diff
@@ -229,6 +234,7 @@ export const cyContent = {
   email: 'E-bost',
   Male: 'Gwryw',
   Female: 'Benyw',
+  completeSectionError: '<span class="govuk-error-message">Llenwch yr adran hon</span>',
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/check-your-answers/content.ts:237]: Don't have html's in translations

File: src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts [hunk 3]
```diff
@@ -545,7 +553,11 @@ export const ApplicantDetails = (
...
+        value: '',
+        valueHtml: !_.isEmpty(sessionApplicantData[applicant]['refugeConfidentialityC8Form'])
+          ? sessionApplicantData[applicant]['refugeConfidentialityC8Form']?.['document_filename']
+          : translation('completeSectionError', language),
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts:560]: construct the span for error

File: src/main/steps/common/confirm-contact-details/checkanswers/ConfirmContactDetailsPostController.ts [hunk 4]
```diff
@@ -84,24 +87,28 @@ export const saveAndRedirectContactDetailsAndPreference = async (
...
-      );
-      mapDataInSession(req.session.userCase, user.id);
-      req.session.userCase.citizenUserAddressText = (await setAddressFields(req)).citizenUserAddressText;
-      req.session.save(() => {
-        const redirectUrl = getRedirectUrl(partyType, req);
-        res.redirect(redirectUrl);
-      });
-    } catch (error) {
-      throw new Error('ConfirmContactDetailsPostController - Case could not be updated.');
+    if (userCase?.isCitizenLivingInRefuge === YesOrNo.YES && _.isEmpty(userCase?.refugeDocument)) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/confirm-contact-details/checkanswers/ConfirmContactDetailsPostController.ts:90]: have a method called isMandatoryFieldsFilled and then allow to submit if the return value is true.

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts (hunk 1 above)
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts:556]: anchorReference

File: src/main/steps/c100-rebuild/check-your-answers/lib/lib.ts [hunk 5]
```diff
@@ -65,6 +69,9 @@ export const getSectionSummaryList = (rows: SummaryListRow[], content: PageConte
                   href: changeUrl, //
                   text: content.change as string,
                   visuallyHiddenText: `${item.key}`,
+                  attributes: {
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/check-your-answers/lib/lib.ts:72]: add id only when achorref is provided

File: src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts (hunk 1 above)
Comment (hcarslaw) [reply src/main/steps/c100-rebuild/check-your-answers/mainUtil.ts]: Changed to c8RefugeDocument-applicant-{applicantIndex} or c8RefugeDocument-otherPerson-{otherPersonIndex}

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**